mkosi/__init__.py

```diff
@@ -1873,111 +1873,92 @@ def make_rpm_list(args: MkosiArgs, packages: Set[str], do_run_build_script: bool
     return packages
 
 
-def clean_dnf_metadata(root: Path, always: bool) -> None:
-    """Remove dnf metadata if /bin/dnf is not present in the image
+def clean_paths(
+        root: Path,
+        globs: List[str],
+        tool: str,
+        always: bool) -> None:
+    """Remove globs under root if always or if tool is not found under root."""
 
-    If dnf is not installed, there doesn't seem to be much use in
-    keeping the dnf metadata, since it's not usable from within the
-    image anyway.
-    """
-    paths = [
-        root / "var/lib/dnf",
-        *root.glob("var/log/dnf.*"),
-        *root.glob("var/log/hawkey.*"),
-        root / "var/cache/dnf",
-    ]
+    tool = root / tool.lstrip('/')
+    cond = always or not os.access(tool, os.F_OK, follow_symlinks=False)
 
-    cond = always or not os.access(root / "bin/dnf", os.F_OK, follow_symlinks=False)
+    paths: List[Path] = list(itertools.chain.from_iterable(
```

Comment:
Ditto for `Sequence` instead of `List`

Reply:
Also maybe add a function `flatten` for this? Makes it a bit more readable I think

Reply:
> Ditto for `Sequence` instead of `List`

`list` will return a list, no need to be overly general ;)

Reply:
With the flatten function, it wouldn't matter either way since type inference would do its thing